src/unix_dgram.cc

```diff
@@ -108,8 +108,13 @@ void OnRecv(SocketContext* sc) {
     err = -errno;
   } else {
     argv[1] = Nan::CopyBuffer(scratch, err).ToLocalChecked();
-    if (u_addr.sun.sun_path[0] != '\0') {
-      argv[2] = Nan::New<String>(u_addr.sun.sun_path).ToLocalChecked();
+    if (msg.msg_namelen > 0) {
+      if (u_addr.sun.sun_path[0] != '\0') {
+        argv[2] = Nan::New<String>(u_addr.sun.sun_path).ToLocalChecked();
+      } else {
+        argv[2] = Nan::New<String>(u_addr.sun.sun_path,
+                    msg.msg_namelen-sizeof(sa_family_t)).ToLocalChecked();
+      }
```

Comment:
I would add a comment here explaining that abstract sockets starts with `\0` or something like that. (If you don't feel like it may not be not that important, but I think it can be helpful for people looking at the code)